core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -28,24 +28,32 @@ oppia.directive('supplementalCard', [
         '/pages/exploration_player/' +
         'supplemental_card_directive.html'),
       controller: [
-        '$scope', '$window', 'oppiaPlayerService',
-        'playerPositionService', 'playerTranscriptService',
-        'ExplorationObjectFactory', 'windowDimensionsService',
-        'CONTENT_FOCUS_LABEL_PREFIX', 'TWO_CARD_THRESHOLD_PX',
-        'EVENT_ACTIVE_CARD_CHANGED', 'CONTINUE_BUTTON_FOCUS_LABEL',
+        '$scope', '$timeout', '$window', 'HintManagerService',
+        'oppiaPlayerService', 'playerPositionService',
+        'playerTranscriptService', 'ExplorationObjectFactory',
+        'windowDimensionsService', 'CONTENT_FOCUS_LABEL_PREFIX',
+        'TWO_CARD_THRESHOLD_PX', 'EVENT_ACTIVE_CARD_CHANGED',
+        'CONTINUE_BUTTON_FOCUS_LABEL',
         function(
-          $scope, $window, oppiaPlayerService,
-          playerPositionService, playerTranscriptService,
-          ExplorationObjectFactory, windowDimensionsService,
-          CONTENT_FOCUS_LABEL_PREFIX, TWO_CARD_THRESHOLD_PX,
-          EVENT_ACTIVE_CARD_CHANGED, CONTINUE_BUTTON_FOCUS_LABEL) {
+          $scope, $timeout, $window, HintManagerService,
+          oppiaPlayerService, playerPositionService,
+          playerTranscriptService, ExplorationObjectFactory,
+          windowDimensionsService, CONTENT_FOCUS_LABEL_PREFIX,
+          TWO_CARD_THRESHOLD_PX, EVENT_ACTIVE_CARD_CHANGED,
+          CONTINUE_BUTTON_FOCUS_LABEL) {
           var updateActiveCard = function() {
             var index = playerPositionService.getActiveCardIndex();
             if (index === null) {
               return;
             }
             $scope.activeCard = playerTranscriptService.getCard(index);
             $scope.clearHelpCard();
+            HintManagerService.disableHintButtonTemporarily();
+
+            HintManagerService.init(oppiaPlayerService.getInteraction(
+              $scope.activeCard.stateName).hints);
+
+            $scope.currentInteractionHints = HintManagerService.getHints();
```

Comment:
You don't need this. You just use the length and that remains the same throughout the state, so just get the length once, store it in a $scope var, and refer to that throughout.

This directive is in charge of displaying the card, it should not know about the internal hint structure or stuff it isn't displaying. That's what the hint service is for.